src/sublime_aio.py

```diff
@@ -82,6 +83,24 @@ def shutdown():
     loop.close()
 
 
+@contextmanager
+def __sync_lock(lock: asyncio.Lock):
+    """Access an async lock from sync land in a blocking way.
+
+    Because the lock is not thread-safe,
+    we ensure to invoke its methods on our thread that runs the event loop.
+    """
+    if not __loop:
+        return
+    asyncio.run_coroutine_threadsafe(lock.acquire(), __loop)
```

Comment:
With regards to the overhead `run_coroutine_threadsafe` causes, I don't like the idea of calling itmore than absolutely needed, especially in possibly heavy duty functions like a debouncer.

Reply:
Dropped the lock commit so it can be discussed separately.